pkg/vmprovider/providers/vsphere2/vmprovider_vm.go

```diff
@@ -1106,56 +1114,12 @@ func (vs *vSphereVMProvider) vmUpdateGetArgs(
 		}
 	}
 
-	var vmImageStatus *vmopv1.VirtualMachineImageStatus
-	// Only get VM image when this is the VM first boot.
-	if isVMFirstBoot(vmCtx) {
-		var err error
-		_, _, vmImageStatus, err = GetVirtualMachineImageSpecAndStatus(vmCtx, vs.k8sClient)
-		if err != nil {
-			return nil, err
-		}
-
-		// The only use of this is for the global JSON_EXTRA_CONFIG to set the image name.
-		// The global extra config should only be set during first boot.
-		// TODO: We can just finally kill this with the demise of old gce2e?
-		renderTemplateFn := func(name, text string) string {
-			t, err := template.New(name).Parse(text)
-			if err != nil {
-				return text
-			}
-			b := strings.Builder{}
-			if err := t.Execute(&b, vmImageStatus); err != nil {
-				return text
-			}
-			return b.String()
-		}
-		extraConfig := make(map[string]string, len(vs.globalExtraConfig))
-		for k, v := range vs.globalExtraConfig {
-			extraConfig[k] = renderTemplateFn(k, v)
-		}
-		updateArgs.ExtraConfig = extraConfig
-
-		// Enabling the defer-cloud-init extraConfig key for V1Alpha1Compatible images defers cloud-init from running on first boot
-		// and disables networking configurations by cloud-init. Therefore, only set the extraConfig key to enabled
-		// when the vmMetadata is nil or when the transport requested is not CloudInit.
-		updateArgs.VirtualMachineImageV1Alpha1Compatible =
-			conditions.IsTrueFromConditions(vmImageStatus.Conditions, vmopv1.VirtualMachineImageV1Alpha1CompatibleCondition)
-	}
-
 	updateArgs.ConfigSpec = virtualmachine.CreateConfigSpec(
 		vmCtx,
 		vmClassConfigSpec,
 		&updateArgs.VMClass.Spec,
-		vmImageStatus,
```

Comment:
The image status is used for the firmware and HW versions fallback. Neither is something we care about in the update path. This still kind of stinks and we need to keep hacking away at what's left over of the old session code.

Reply:
If we add `spec.firmware` the way we have `spec.minHardwareVersion`, do we need to keep anything? Can we not just copy the values to those fields if they are empty?

Reply:
There is a `vmoperator.vmware.com/firmware` annotation. In the update path we don't use the those fields from the ConfigSpec created here, so what values don't really matter. Both `spec.minHardwareVersion` and the firmware annotation are what's used in the update path. I don't think a `spec.firmware` gains us anything specific for here, and IDK if that's used/cared enough about for a spec field. All this is still a confusing mix of pre and post CL DeployOVF w/ ConfigSpec work that needs to kept being chipped away at for resize work, and our sanity.